vizro-core/src/vizro/_vizro.py

```diff
@@ -36,6 +37,8 @@ def __init__(self, **kwargs):
                 [Dash documentation](https://dash.plotly.com/reference#dash.dash) for possible arguments.
 
         """
+        self._undefined_captured_callables: list[str] = []
```

Comment:
```suggestion
        self._undefined_captured_callables: set[str] = {}
```

Very minor suggestion.